### DISCUSSION TITLE
refresh token failed: HTTP status client error (400 Bad Request): The input parameter client_id is not valid

### DISCUSSION THREAD

**Discussion by HelimLee** [messense/aliyundrive-webdav#210] [Q&A] [answered]
### 问题描述

<html>
<body>
<div class="x-grid3-row x-grid3-row-over x-grid3-row-selected" id="ext-gen16799" style="margin: 0px; padding: 0px; border-width: 0px 0px 1px; border-style: dotted dotted solid; border-color: rgb(220, 230, 240) rgb(220, 230, 240) rgb(235, 240, 245); border-image: initial; cursor: default; width: 686px; background-image: none; background-position: left top; background-size: initial; background-repeat: repeat-x; background-attachment: initial; background-origin: initial; background-clip: initial; background-color: rgb(222, 236, 252) !important; color: rgb(80, 90, 100); font-size: 12px; height: 24px;">

2021-12-20T14:10:25.209990Z ERROR aliyundrive_webdav::drive: refresh token failed: HTTP status client error (400 Bad Request) for url (https://websv.aliyundrive.com/token/refresh): {"code":"InvalidParameter.ClientId","message":"The input parameter client_id is not valid. Please refer to document.","requestId":null} | stdout
-- | --

</div><div class="x-grid3-row" id="ext-gen16805" style="margin: 0px; padding: 0px; border-width: 0px 0px 1px; border-style: solid; border-color: rgb(255, 255, 255) rgb(237, 237, 237) rgb(235, 240, 245); border-image: initial; cursor: default; width: 686px; background-color: transparent; color: rgb(80, 90, 100); font-size: 12px; height: 24px;"><table class="x-grid3-row-table" border="0" cellspacing="0" cellpadding="0" style="table-layout: fixed; width: 686px; height: 24px;"><tbody><tr role="row" id="ext-gen16772" aria-haspopup="true" aria-selected="false"><td class="x-grid3-col x-grid3-cell x-grid3-td-time x-grid3-cell-first " role="gridcell" tabindex="-1" id="ext-gen16769" aria-readonly="true" style="margin: 0px; padding: 0px 1px; outline: 0px; line-height: 13px; vertical-align: top; font-weight: 400; font-size: 12px; height: 24px; color: rgb(80, 90, 100); font-family: verdana, arial, tahoma, helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; width: 108px;"><div class="x-grid3-cell-inner x-grid3-col-time x-unselectable" unselectable="on" style="margin: 0px; padding: 0px 8px; user-select: none; overflow: hidden; text-overflow: ellipsis; white-space: nowrap; color: rgb(80, 90, 100); font-size: 12px; width: auto; line-height: 24px;">14:10:25</div></td><td class="x-grid3-col x-grid3-cell x-grid3-td-1 " role="gridcell" tabindex="-1" id="ext-gen16770" aria-readonly="true" style="margin: 0px; padding: 0px 1px; outline: 0px; line-height: 13px; vertical-align: top; font-weight: 400; font-size: 12px; height: 24px; color: rgb(80, 90, 100); font-family: verdana, arial, tahoma, helvetica, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; width: 461px;"><div class="x-grid3-cell-inner x-grid3-col-1 x-unselectable" unselectable="on" style="margin: 0px; padding: 0px 8px; user-select: none; overflow: hidden; text-overflow: ellipsis; white-space: nowrap; color: rgb(80, 90, 100); font-size: 12px; width: auto; line-height: 24px;"><br class="Apple-interchange-newline">
</body>
</html>

### 重现步骤

正常启动时

### 版本

1.0.0

### 运行平台

Docker Synology

### 日志

<html>
<body>

  | <!DOCTYPE html>
-- | --
  | <html><head>
  | <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
  | <style>body{font-family:"Lucida Console",Consola,Monaco,Monospace,"Courier New",Courier,Serif;}h2{text-align:center;}table{width:95%;border-collapse:collapse;}table,th,td{border:1px solid black;}td{white-space:nowrap;text-align:left;}</style>
  | </head>
  | <body>
  | <h2>aliyundrive-webdav</h2>
  | <table>
  | <tr><th>date</th><th>stream</th><th>content</th></tr>
  | <tr><td>2021-12-20&nbsp;14:10:35</td><td>stderr</td><td>Error:&nbsp;initialize&nbsp;aliyundrive&nbsp;client&nbsp;failed
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:35</td><td>stdout</td><td>[2m2021-12-20T14:10:35.417258Z[0m&nbsp;[31mERROR[0m&nbsp;[2maliyundrive_webdav::drive[0m[2m:[0m&nbsp;refresh&nbsp;token&nbsp;failed:&nbsp;HTTP&nbsp;status&nbsp;client&nbsp;error&nbsp;(400&nbsp;Bad&nbsp;Request)&nbsp;for&nbsp;url&nbsp;(https://websv.aliyundrive.com/token/refresh):&nbsp;{"code":"InvalidParameter.ClientId","message":"The&nbsp;input&nbsp;parameter&nbsp;client_id&nbsp;is&nbsp;not&nbsp;valid.&nbsp;Please&nbsp;refer&nbsp;to&nbsp;document.","requestId":null}
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:31</td><td>stderr</td><td>Error:&nbsp;initialize&nbsp;aliyundrive&nbsp;client&nbsp;failed
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:31</td><td>stdout</td><td>[2m2021-12-20T14:10:31.841261Z[0m&nbsp;[31mERROR[0m&nbsp;[2maliyundrive_webdav::drive[0m[2m:[0m&nbsp;refresh&nbsp;token&nbsp;failed:&nbsp;HTTP&nbsp;status&nbsp;client&nbsp;error&nbsp;(400&nbsp;Bad&nbsp;Request)&nbsp;for&nbsp;url&nbsp;(https://websv.aliyundrive.com/token/refresh):&nbsp;{"code":"InvalidParameter.ClientId","message":"The&nbsp;input&nbsp;parameter&nbsp;client_id&nbsp;is&nbsp;not&nbsp;valid.&nbsp;Please&nbsp;refer&nbsp;to&nbsp;document.","requestId":null}
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:28</td><td>stderr</td><td>Error:&nbsp;initialize&nbsp;aliyundrive&nbsp;client&nbsp;failed
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:28</td><td>stdout</td><td>[2m2021-12-20T14:10:28.412963Z[0m&nbsp;[31mERROR[0m&nbsp;[2maliyundrive_webdav::drive[0m[2m:[0m&nbsp;refresh&nbsp;token&nbsp;failed:&nbsp;HTTP&nbsp;status&nbsp;client&nbsp;error&nbsp;(400&nbsp;Bad&nbsp;Request)&nbsp;for&nbsp;url&nbsp;(https://websv.aliyundrive.com/token/refresh):&nbsp;{"code":"InvalidParameter.ClientId","message":"The&nbsp;input&nbsp;parameter&nbsp;client_id&nbsp;is&nbsp;not&nbsp;valid.&nbsp;Please&nbsp;refer&nbsp;to&nbsp;document.","requestId":null}
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:25</td><td>stderr</td><td>Error:&nbsp;initialize&nbsp;aliyundrive&nbsp;client&nbsp;failed
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:25</td><td>stdout</td><td>[2m2021-12-20T14:10:25.209990Z[0m&nbsp;[31mERROR[0m&nbsp;[2maliyundrive_webdav::drive[0m[2m:[0m&nbsp;refresh&nbsp;token&nbsp;failed:&nbsp;HTTP&nbsp;status&nbsp;client&nbsp;error&nbsp;(400&nbsp;Bad&nbsp;Request)&nbsp;for&nbsp;url&nbsp;(https://websv.aliyundrive.com/token/refresh):&nbsp;{"code":"InvalidParameter.ClientId","message":"The&nbsp;input&nbsp;parameter&nbsp;client_id&nbsp;is&nbsp;not&nbsp;valid.&nbsp;Please&nbsp;refer&nbsp;to&nbsp;document.","requestId":null}
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:22</td><td>stderr</td><td>Error:&nbsp;initialize&nbsp;aliyundrive&nbsp;client&nbsp;failed
  | </td></tr>
  | <tr><td>2021-12-20&nbsp;14:10:22</td><td>stdout</td><td>[2m2021-12-20T14:10:22.417652Z[0m&nbsp;[31mERROR[0m&nbsp;[2maliyundrive_webdav::drive[0m[2m:[0m&nbsp;refresh&nbsp;token&nbsp;failed:&nbsp;HTTP&nbsp;status&nbsp;client&nbsp;error&nbsp;(400&nbsp;Bad&nbsp;Request)&nbsp;for&nbsp;url&nbsp;(https://websv.aliyundrive.com/token/refresh):&nbsp;{"code":"InvalidParameter.ClientId","message":"The&nbsp;input&nbsp;parameter&nbsp;client_id&nbsp;is&nbsp;not&nbsp;valid.&nbsp;Please&nbsp;refer&nbsp;to&nbsp;document.","requestId":null}
  | </td></tr>
  | </table></body></html>

</body>
</html>

**Comment by messense** [comment, marked as answer]
重新在网页端获取 refresh token 试试。

**Comment by HelimLee** [comment]
可以了。之前我用Onemanager是从手机log里面获取的refresh_Token,原来这两个还不一样啊
